src/fights.ts

```diff
@@ -63,95 +64,99 @@ import { withStash } from "./stash";
+            Macro.trySkill("Lecture on Relativity")
+              .tryItem($item`Spooky Putty sheet`)
               .meatKill(),
-            () => use($item`photocopied monster`)
+            () => use($item`shaking 4-d camera`)
           );
+          set("_garbo_professorLecturesUsed", true);
+        } else if (!get("_photocopyUsed")) {
+          withMacro(Macro.tryItem($item`Spooky Putty sheet`).meatKill(), () => {
+            use($item`photocopied monster`);
+          });
+          set("_garbo_professorLecturesUsed", true);
         }
 
-        if (
-          getCounters("Digitize Monster", 0, 100).trim() === "" &&
-          get("_mushroomGardenFights") === 0
-        ) {
-          if (have($item`packet of mushroom spores`)) use($item`packet of mushroom spores`);
-          // adventure in mushroom garden to start digitize timer.
-          freeFightOutfit();
+        let puttyCount = 1;
+        while (availableAmount($item`Spooky Putty monster`) > 0 && puttyCount <= 5) {
           useFamiliar(meatFamiliar());
-          adventureMacro($location`Your Mushroom Garden`, Macro.meatKill());
+          meatOutfit(true);
+          withMacro(
+            Macro.externalIf(
+              get("spookyPuttyCopiesMade") < 5 && puttyCount < 5,
+              Macro.item($item`Spooky Putty sheet`)
+            ).meatKill(),
+            () => use($item`Spooky Putty monster`)
+          );
+          puttyCount++;
         }
-
-        // Second round of prof copies with familiar weight on.
-        freeFightMood().execute(20);
-        useFamiliar($familiar`Pocket Professor`);
-        maximizeCached(["Familiar Weight"], { forceEquip: $items`Pocket Professor memory chip` });
-        withMacro(
-          Macro.trySkill("Lecture on Relativity")
-            .tryItem($item`Spooky Putty sheet`)
-            .meatKill(),
-          () => use($item`shaking 4-d camera`)
-        );
-        set("_garbo_professorLecturesUsed", true);
-      } else if (!get("_photocopyUsed")) {
-        withMacro(Macro.tryItem($item`Spooky Putty sheet`).meatKill(), () => {
-          use($item`photocopied monster`);
-        });
-        set("_garbo_professorLecturesUsed", true);
-      }
-
-      let puttyCount = 1;
-      while (availableAmount($item`Spooky Putty monster`) > 0 && puttyCount <= 5) {
-        useFamiliar(meatFamiliar());
-        meatOutfit(true);
-        withMacro(
-          Macro.externalIf(
-            get("spookyPuttyCopiesMade") < 5 && puttyCount < 5,
-            Macro.item($item`Spooky Putty sheet`)
-          ).meatKill(),
-          () => use($item`Spooky Putty monster`)
-        );
-        puttyCount++;
-      }
-      set("spookyPuttyCopiesMade", 5);
-    });
+        set("spookyPuttyCopiesMade", 5);
+      });
+    }
   }
+  if (have($item`Packet of tall grass seeds`)) use($item`Packet of tall grass seeds`);
+  if (have($item`unwrapped retro superhero cape`)) cliExecute("retrocape robot kill");
```

Comment:
these checks should both be in the main adventure loop, not here

Reply:
Resolved as of b9f33ec